src/preact.d.ts

```diff
@@ -52,7 +52,7 @@ declare namespace preact {
 
 		linkState:(name:string) => (event: Event) => void;
 
-		setState(state:StateType, opts?:any):void;
+		setState(state:Partial<StateType>, opts?:any):void;
```

Comment:
Awesome. Using Partial is the way to go here. Just wish there was a way for libraries to specify what version of TypeScript their definition files require to compile.